File: Model/MetricFilters/SearchDTO.cs
using LightMethods.Survey.Models.FilterValues;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LightMethods.Survey.Models.MetricFilters
{
    public class SearchDTO
    {
        public Guid ProjectId { get; set; }

        public List<Guid> FormTemplateIds { get; set; }

        public string Term { get; set; }

        public DateTime? StartDate { get; set; }

        public DateTime? EndDate { get; set; }

        public List<FilterValue> FilterValues { get; set; }

        public SearchDTO()
        {
            this.FilterValues = new List<FilterValue>();
        }
    }
}

File: WebApp/wwwroot/scripts/domain/searchDTO.ts
module App.Models {
    "use strict";

    export class SearchDTO {
        public projectId: string;
        public formTemplateIds: string[];
        public term: string;
        public startDate?: Date;
        public endDate?: Date;
        public filterValues: Models.IFilterValue[]

        constructor() {
            this.projectId = '';
            this.formTemplateIds = [];
            this.term = '';
            this.startDate = null;
            this.endDate = null;
            this.filterValues = [];
        }
    }
}